### PR TITLE
JAMES-3319 Effective blob deletion for RabbitMQMailQueue

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java [hunk 1]
```diff
@@ -174,6 +179,78 @@ void browseShouldReturnCurrentlyEnqueuedMailFromAllSlices() throws Exception {
...
+            String name1 = "myMail1";
+            getMailQueue().enQueue(defaultMail()
+                .name(name1)
+                .sender(SENDER)
+                .build());
+
+            getManageableMailQueue().remove(ManageableMailQueue.Type.Sender, SENDER.asString());
+
+            assertThat(cassandra.getConf().execute(select().from(BlobTables.DefaultBucketBlobTable.TABLE_NAME)))
+                .isEmpty();
```
Comment (rouazana) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java:251]: do you have a test checking that if there are 2 mails in the queue, and we remove one, the matching blobs are removed but not the remaining ones?

**Comment by chibenwa** [comment]
```
[698c1321fc8478ab9b4b7d2bd3b2992444199fec] [ERROR] Failures: 
[698c1321fc8478ab9b4b7d2bd3b2992444199fec] [ERROR]   JMSCacheableMailQueueTest.removeByNameShouldRemoveSpecificEmail 
[698c1321fc8478ab9b4b7d2bd3b2992444199fec] Expecting:
[698c1321fc8478ab9b4b7d2bd3b2992444199fec]   <[]>
[698c1321fc8478ab9b4b7d2bd3b2992444199fec] to contain exactly (and in same order):
[698c1321fc8478ab9b4b7d2bd3b2992444199fec]   <["name1"]>
[698c1321fc8478ab9b4b7d2bd3b2992444199fec] but could not find the following elements:
[698c1321fc8478ab9b4b7d2bd3b2992444199fec]   <["name1"]>
```

unrelated

**Comment by chibenwa** [comment]
> So these test are not valid for a deduplicating blob store?

Now there is.

File: server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueView.java [hunk 2]
```diff
@@ -133,15 +143,18 @@ private long browseThenDelete(DeleteCondition deleteCondition) {
         return cassandraMailQueueBrowser.browseReferences(mailQueueName)
             .map(EnqueuedItemWithSlicingContext::getEnqueuedItem)
             .filter(deleteCondition::shouldBeDeleted)
-            .flatMap(mailReference -> cassandraMailQueueMailDelete.considerDeleted(mailReference.getEnqueueId(), mailQueueName))
+            .flatMap(mailReference -> cassandraMailQueueMailDelete.considerDeleted(mailReference.getEnqueueId(), mailQueueName)
```
Comment (remk) [inline server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueView.java:146]: Should we add a max concurrency parameter on the flatMap ?

**Comment by chibenwa** [comment]
`[ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:3.4.4:testCompile (scala-test-compile) on project event-sourcing-pojo: wrap: org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]`

test this please

**Comment by chibenwa** [comment]
```
[9bf70e50272512b4a039abdcae34992429e1d09a] [ERROR]   RabbitMQWorkQueuePersistenceTest.submittedMessageShouldSurviveRabbitMQRestart » ShutdownSignal
```

unrelated

test this please

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**